pilot/pkg/bootstrap/server.go

```diff
@@ -1108,28 +1108,7 @@ func (s *Server) maybeCreateCA(caOpts *caOptions) error {
 	return nil
 }
 
-// StartCA starts the CA or RA server if configured.
-func (s *Server) startCA(caOpts *caOptions) {
-	if s.CA == nil && s.RA == nil {
-		return
-	}
-	s.addStartFunc(func(stop <-chan struct{}) error {
-		grpcServer := s.secureGrpcServer
-		if s.secureGrpcServer == nil {
-			grpcServer = s.grpcServer
-		}
-		// Start the RA server if configured, else start the CA server
-		if s.RA != nil {
-			log.Infof("Starting RA")
-			s.RunCA(grpcServer, s.RA, caOpts)
-		} else if s.CA != nil {
-			log.Infof("Starting IstioD CA")
-			s.RunCA(grpcServer, s.CA, caOpts)
-		}
-		return nil
-	})
-}
-
+// TODO: remove it, currently used to indicate whether to start namespace controller
 func (s *Server) fetchCARoot() map[string]string {
 	if s.isDisableCa() {
 		return map[string]string{
```

Comment:
Not very familiar with this code but I am not certain that we actually respect the `string(s.RA.GetCAKeyCertBundle().GetRootCertPem())`  part?

We probably need more test coverage for this code..

Reply:
Yes, this code is very tricky, i want to cleanup it later